### PR TITLE
[WFLY-11034] Implemented two attributes: "wildfly-grpc-server-port" a…

### PR DESCRIPTION
**Pull request by ronsigal**
…nd "wildfly-grpc-server-host"

### CODE REVIEW

**Comment by ronsigal** [comment]
This is a first step in extending the grpc subsystem. There are two read/write attributes that appear in CLI.

@hpehl, are there any pull request conventions I should know about?

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java:84]: Upfront warning. I'm going to be nit-picky :)

I think these should be removed. There is not a strong reason to differentiate them.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java:40]: Since everything is in the same package, the visibility should be reduced.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: No tabs please :)

Comment (jamezp) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: These names should be more generic like "server-port". There is no real reason for the "wildfly-grpc".

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: All of the below should be deletable with a `PersistentResourceXMLParser`.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfigService.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfigService.java:36]: I don't see why we'd need a service here. The start/stop don't really do anything.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfig.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfig.java:43]: IMO we should not need this. We should be setting values directly on the `GrpcServerService`.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcParamHandler.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcParamHandler.java:33]: We should just use the `ReloadRequiredWriteAttributeHandler`. Unless the service itself can be restarted.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcExtension.java
Comment (jamezp) [inline subsystem/src/main/java/org/wildfly/extension/grpc/GrpcExtension.java:73]: We should just use the `ReloadRequiredWriteAttributeHandler`. Unless the service itself can be restarted.

**Review by jamezp — CHANGES_REQUESTED** [review]
I mentioned on the comments, I'm being picky here and @hpehl can definitely override me. However, we should be using the more modern subsystem modeling and MSC services like this is already setup with.

**Comment by hpehl** [comment]
@ronsigal I've set up some initial rules for PRs in the settings: 
- Require a PR before merging
- Require review/approval from a code owner

If that's too awkward, feel free to adjust them.

**Review by hpehl — APPROVED** [review]
@jamezp I'm all ok with your suggestions. You have way more experience in subsystem development than I have. My code is a copy & paste from other subsystems and is probably not up to date with the latest APIs.

**Comment by hpehl** [comment]
@ronsigal How should we proceed? Can you resolve the requested changes?

**Comment by ronsigal** [comment]
Heyyy @jamezp,

Thank you for looking at this stuff. I copy and pasted directly from what the jaxrs subsystem, which I worked on in 2019, I think. Maybe it's out of date by now.  I'll respond to the individual comments, but the main thing is, I just don't understand the sentence

>we should be using the more modern subsystem modeling and MSC services like this is already setup with.

My main reference for jaxrs was "Extending WildFly" (https://docs.wildfly.org/27/Extending_WildFly.html, or whatever the version was at the time), which, I see, mentions MSC. Where should I be looking?

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java:40]: Ok

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: > No tabs please :)

Yes, I was meaning to ask where to find the eclipse format/template files ...

> These names should be more generic like "server-port".

Ok

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: > All of the below should be deletable with a PersistentResourceXMLParser.

I see that PersistentResourceXMLParser is mentioned in "Extending Wildfly", so there's a minimal chance that I had a good reason for not using PersistentResourceXMLParser. More likely, it's just ignorance. I'll look at it.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java
Comment (darranl) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: @ronsigal do you use Eclipse, I thought I was the only one left - two of us use Eclipse !!!

I always get my settings from the WildFly Core repo under ide

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfigService.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfigService.java:36]: Mmmm, no idea ...

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfig.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfig.java:43]: Ok ...

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcParamHandler.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcParamHandler.java:33]: Ok ...

**Comment by ronsigal** [comment]
> @ronsigal How should we proceed? Can you resolve the requested changes?

Heyyy @hpehl,

I will do my best to understand @jamezp's comments and resolve them. Clearly, I have more to learn ;-)

-Ron

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java
Comment (jamezp) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: The `PersistentResourceXMLParser` is the "new" way of doing subsystems. I will admit I have NOT used it yet, but I looked at other subsystems that do which was the reason I think the below was not needed. I'm definitely happy to help look though, so feel free to ping me on chat. I'd like to learn how to use it as well :)

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfigService.java
Comment (jamezp) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcServerConfigService.java:36]: Here we're just storing a configuration which is something I don't really think we need. We can interact directly with the `GrpcServerService` which, IMO, would be the preferred way.

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: @darranl, I'm thinking back to when Bill Burke referred to Eclipse as "open source caca" ;-)

Just don't make me learn Emacs !!

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: Ok, thanks, @jamezp. Reading about PersistentResourceXMLParser will be my next step. Along with deleting the tabs. ;-)

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java
Comment (hpehl) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: @ronsigal The Eclipse format rules are under [build-config/src/main/resources/etc/eclipse-formatter-config.xml](https://github.com/wildfly-extras/wildfly-grpc-feature-pack/blob/main/build-config/src/main/resources/etc/eclipse-formatter-config.xml)

Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: vi, baby!!

I'm guessing I learned vi in 1983, when I got my first Unix account.  I'm reminded of one of my greatest accomplishments in those days, which was spending several hours figuring out how a colleague of mine could send email from Brooklyn College to another researcher at the University of Michigan. You had to map out a sequence of intervening machines. E.g., Brooklyn College -> a machine at NYU -> a machine at University of Pennsylvania ... . Those were the days.  ;-)

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: Ok, I see:
```
    static {
        xmlDescription = builder(GrpcSubsystemDefinition.INSTANCE.getPathElement(), GrpcSubsystemSchema.GRPC_1_0.getUriString())
                .addAttributes(
                        GrpcAttribute.WILDFLY_GRPC_SERVER_PORT,
                        GrpcAttribute.WILDFLY_GRPC_SERVER_HOST)
                .build();
    }
```
is all you need in GrpcSubsystemParser_1_0. Wow!

Comment (jamezp) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: Yeah, it's pretty nice eh?

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java:40]: Done

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcAttribute.java:84]: Removed

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcConstants.java:25]: Tabs, names fixed

File: subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java
Comment (ronsigal) [reply subsystem/src/main/java/org/wildfly/extension/grpc/GrpcSubsystemParser_1_0.java:54]: Fixed

**Pull request merged**
